app/models/user_access.rb

```diff
@@ -67,6 +95,48 @@ def can_access_record?(resources, record)
     resources.exists?
   end
 
+  def can_create?(model, record)
+    case model
+    when :facility
+      can_create_record?(:facility, record, accessible_facility_groups(:manage))
+    when :facility_group
+      can_create_record?(:facility_group, record, accessible_organizations(:manage))
+    when :organization
+      can_create_record?(:organization, record)
+    else
+      raise ArgumentError, "Access to #{model} is unsupported."
+    end
+  end
+
+  def can_create_record?(model, record, resources = nil)
+    case model
+    when :facility
+      resources&.find_by_id(record.facility_group_id).present?
+    when :facility_group
+      resources&.find_by_id(record.organization_id).present?
+    when :organization
+      user.power_user?
+    else
+      raise ArgumentError, "Can't create #{model}."
+    end
+  end
+
+  def can_perform_action?(action, model, record = nil)
```

Comment:
This is only called when record is `nil` right? Why does it need the third argument?